### PR TITLE
Made function image_name

### PR DESCRIPTION
**Pull request by MrRaccoonUA**
# What does this pull request do?
Made a function that generates a filename from the resulting product name, removing the separator characters

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -41,16 +41,16 @@ def extractor(link, prefix):
     https://i5. - start of the Walmart link
     """
 
-    def target_extractor(TARGET_DEFAULT_SIZE, TARGET_MAX_SIZE, target_prefix):
+    def target_extractor(DEFAULT_SIZE, MAX_SIZE, target):
```
Comment (aartiukh) [inline main.py:44]: I believe we should not use UPPER_CASE for input args

File: main.py [hunk 2]
```diff
@@ -88,9 +98,9 @@ def walmart_extractor(walmart_prefix):
         detail = api_search(code, API_URL_GET)
         if not detail:
             continue
-        print(type(detail.Images_Links))
         extracted_link = link_extractor(detail.UPC, detail.Images_Links)
-        print(extracted_link)
+        file_name = image_name(detail.UPC, detail.Product_Name)
+        print(file_name)
```
Comment (aartiukh) [inline main.py:103]: Remove this

File: main.py (hunk 1 above)
Comment (MrRaccoonUA) [reply main.py]: Fixed

File: main.py (hunk 2 above)
Comment (MrRaccoonUA) [reply main.py]: Fixed

**Review by aartiukh — APPROVED** [review]

File: main.py [hunk 3]
```diff
@@ -11,16 +11,16 @@
                     datefmt='%Y-%m-%d %I:%M:%S')
 
 
-def api_search(code, API_URL):
+def api_search(upc, API_URL):
```
Comment (aartiukh) [inline main.py:14]: Add documentation

File: main.py [hunk 4]
```diff
@@ -29,10 +29,10 @@ def api_search(code, API_URL):
 
 
 def link_extractor(upc, list_of_links):
-    def extractor(link, prefix):
-        word = len(prefix)
+    def extractor(link, message):
```
Comment (aartiukh) [inline main.py:32]: Add documentation

File: main.py [hunk 5]
```diff
@@ -70,6 +70,16 @@ def walmart_extractor(walmart_prefix):
...
 
 
+def image_name(upc, product_name):
+    File_Name = namedtuple('File_Name', 'file_name')
+    character_search = re.sub(r'[-,?%/®\"]\s|(/)|(\")|[-]', ' ', product_name)
+    delete_spaces = character_search.replace(' ', '_')
+    delete_double_underscore = delete_spaces.replace('__', '_')
+    log_upc = upc + ' - ' + delete_double_underscore
+    logger.info(f'Rename file {log_upc}: OK')
+    return File_Name(delete_double_underscore)
```
Comment (aartiukh) [inline main.py:80]: Add empty line before this one

File: main.py [hunk 6]
```diff
@@ -88,9 +98,8 @@ def walmart_extractor(walmart_prefix):
         detail = api_search(code, API_URL_GET)
         if not detail:
             continue
-        print(type(detail.Images_Links))
         extracted_link = link_extractor(detail.UPC, detail.Images_Links)
-        print(extracted_link)
+        file_name = image_name(detail.UPC, detail.Product_Name)
```
Comment (aartiukh) [inline main.py:102]: `file_name` is unused?

**Review by aartiukh — CHANGES_REQUESTED** [review]

File: main.py (hunk 3 above)
Comment (MrRaccoonUA) [reply main.py:14]: Added

File: main.py (hunk 4 above)
Comment (MrRaccoonUA) [reply main.py:32]: Added

File: main.py (hunk 5 above)
Comment (MrRaccoonUA) [reply main.py:80]: Fixed

File: main.py (hunk 6 above)
Comment (MrRaccoonUA) [reply main.py:102]: Fixed

**Review by aartiukh — APPROVED** [review]

**Pull request merged**